base_report_to_printer/models/printing_server.py

```diff
@@ -43,6 +48,13 @@ def _open_connection(self, raise_on_error=False):
             # (see https://github.com/OpenPrinting/pycups/issues/30)
             cups.setServer(self.address)
             cups.setPort(self.port)
+            if self.user:
```

Comment:
you separate 3 cases, could you please explain this reason.
user/password policy should be set up in the same process, you code suppose that cups will allow to connect with a user and no password. Does cups allow this behaviour ?